nucleus/common/glassfish-api/src/main/java/org/glassfish/api/invocation/ComponentInvocation.java

```diff
@@ -271,6 +275,22 @@ public void setModuleName(String moduleName) {
         this.moduleName = moduleName;
     }
 
+    /**
+     * Returns the unique registration name of application.
+     *
+     * @return
+     */
+    public String getRegistrationName() {
+        if (registrationName != null) {
+            return registrationName;
+        }
+        return getAppName();
```

Comment:
I think this is somewhat suprising to the caller to get the AppName when they are expecting the RegistrationName. IMHO this should be handled by the caller.